standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -7420,6 +7420,64 @@ public boolean delete_partition_column_statistics(String dbName, String tableNam
     return ret;
   }
 
+  @Override
+  public boolean delete_partition_column_statistics_req(DeletePartitionColumnStatisticsRequest req) throws TException {
+    String dbName = req.getDb_name();
+    String tableName = req.getTbl_name();
+    List<String> colNames = req.getCol_names();
+    String partName = req.getPart_name();
+    String engine = req.getEngine();
+    dbName = dbName.toLowerCase();
+    String[] parsedDbName = parseDbName(dbName, conf);
+    tableName = tableName.toLowerCase();
+    if (colNames != null) {
+      for (String colName : colNames) {
+        colName = colName.toLowerCase();
+        startFunction("delete_column_statistics_by_partition",": table=" +
+                TableName.getQualified(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tableName) +
+                " partition=" + partName + " column=" + colName);
+      }
+    }
+    boolean ret = false, committed = false;
+
+    getMS().openTransaction();
+    try {
+      List<String> partVals = getPartValsFromName(getMS(), parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tableName, partName);
+      Table table = getMS().getTable(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tableName);
+      // This API looks unused; if it were used we'd need to update stats state and write ID.
+      // We cannot just randomly nuke some txn stats.
+      if (TxnUtils.isTransactionalTable(table)) {
```

Comment:
why can't we drop the transactional statistics? can we make the partition stats as incorrect in partition params by `StatsSetupConst.COLUMN_STATS_ACCURATE`